docs/subnets/commit-reveal.md

```diff
@@ -1,25 +1,57 @@
+
+The only solution for this is to continuously change demands on miners, requiring them to continuously evolve their behavior to maintain their scoring. Combined with a properly tuned Commit Reveal interval, this will keep validators honest, as well as producing the best models.
+
+## Commit Reveal and Immunity Period
+
+The [Immunity Period](../glossary.md#immunity-period) is the interval (measured in blocks) during which a miner or validator newly registered on a subnet is 'immune' from deregistration due to performance, because their performance is not yet scored. The duration of this period value should always be larger than the Commit Reveal interval, otherwise the immunity period will expire before a given miner's scores are available, and they may be deregistered without having their work counted.
+
+To ensure that the immunity period is always greater than the Commit Reveal interval, we recommend using the following logic when setting the Immunity Period or Commit Reveal interval hyperparameters for a subnet:
+
+```
+new_immunity_period = (new_commit_reveal_interval - old_commit_reveal_interval) + old_immunity_period
+```
+
+See [Subnet Hyperparameters](./subnet-hyperparameters.md).
+
+
+## Commit reveal in detail
+
+When commit reveal is enabled, it works as follows:  
+
+1. A subnet validator sets the weights normally by using [`set_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/set_weights/index.html). 
+
+2. Instead of publishing weights openly, an encrypted hash of these consensus weights is committed to the blockchain, using an internal method called [`commit_weights`](pathname:///python-api/html/autoapi/bittensor/core/extrinsics/commit_weights/index.html).
+
+3. A waiting interval, specified as a number of tempos, elapses. Subnet owners configure this interval with the subnet hyperparameter `commit_reveal_weights_interval`.
 
-The commit reveal feature is designed to combat this weight-copying problem by giving the weight-copiers access only to the old weights. The main idea is this: The consensus weights set by subnet validators are hidden for a certain interval. The weight-copiers would be able to view these weights only after this interval is elapsed. But by then these consensus weights would've become quite old, ineffective and hence would create a loss for the weight-copier using these copied old weights. As a result, weight-copiers would not have advantage over honest subnet validators.
+4. After this interval has elapsed, the unencrypted weights are published to the blockchain.
```

Comment:
Add note about ‘deren’ (?) pulse mechanism for publishing decryption key after the commit reveal interval that unlocks the previously published weight matrix.